File: app/src/main/java/com/artamonov/millionplanets/market/MarketPlanetAdapter.java
package com.artamonov.millionplanets.market;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.artamonov.millionplanets.R;
import com.artamonov.millionplanets.model.PlanetType;
import com.artamonov.millionplanets.model.User;
import com.artamonov.millionplanets.utils.Price;
import com.artamonov.millionplanets.utils.Utils;

public class MarketPlanetAdapter extends RecyclerView.Adapter<MarketPlanetAdapter.ViewHolder> {

    private static DialogListener dialogListener;
    public User user;
    private @PlanetType.AnnotationPlanetType String planetType;

    public MarketPlanetAdapter(User user, String planetType, DialogListener listener) {
        this.user = user;
        this.planetType = planetType;
        dialogListener = listener;
    }

    @NonNull
    @Override
    public MarketPlanetAdapter.ViewHolder onCreateViewHolder(
            @NonNull ViewGroup parent, int viewType) {
        View view =
                LayoutInflater.from(parent.getContext())
                        .inflate(R.layout.market_you_items, parent, false);
        return new MarketPlanetAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MarketPlanetAdapter.ViewHolder holder, int position) {
        holder.resourceName.setText(Utils.INSTANCE.getResourceItemNameById(position));
        holder.resourcePrice.setText(String.valueOf(Price.INSTANCE.getPlayerBuyPrice(position)));
    }

    @Override
    public int getItemCount() {
        return Utils.INSTANCE.getResourceTypeAmountByPlanetType(planetType);
    }

    public interface DialogListener {
        void onDialogCreate(int position);
    }

    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        private final TextView resourceAmount;
        private final TextView resourceName;
        private final TextView resourcePrice;

        ViewHolder(View itemView) {
            super(itemView);
            resourceName = itemView.findViewById(R.id.resource_name);
            resourceAmount = itemView.findViewById(R.id.resource_amount);
            resourcePrice = itemView.findViewById(R.id.resource_price);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            dialogListener.onDialogCreate(getAdapterPosition());
        }
    }
}

File: app/src/main/java/com/artamonov/millionplanets/market/adapter/MarketPlanetAdapter.kt
package com.artamonov.millionplanets.market.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.artamonov.millionplanets.R
import com.artamonov.millionplanets.model.PlanetType.AnnotationPlanetType
import com.artamonov.millionplanets.model.User
import com.artamonov.millionplanets.utils.Price.getPlayerBuyPrice
import com.artamonov.millionplanets.utils.getResourceItemName
import com.artamonov.millionplanets.utils.getResourceTypeAmountByPlanetType
import kotlinx.android.synthetic.main.market_you_items.view.*

class MarketPlanetAdapter(
    var user: User?,
    @field:AnnotationPlanetType private val planetType: String?,
    private var dialogListener: DialogListener
) : RecyclerView.Adapter<MarketPlanetAdapter.ViewHolder>() {

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.market_you_items, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindItem()
    }

    override fun getItemCount(): Int {
        return planetType.getResourceTypeAmountByPlanetType()
    }

    interface DialogListener {
        fun onDialogCreate(position: Int)
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        fun bindItem() {
            itemView.resource_name.text = adapterPosition.getResourceItemName()
            itemView.resource_price.text = getPlayerBuyPrice(adapterPosition).toString()
            itemView.setOnClickListener {
                dialogListener.onDialogCreate(adapterPosition)
            }
        }
    }
}